### PR TITLE
Decompile obj_wturn

### PR DESCRIPTION
**Pull request by quaantuum**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c [hunk 1]
```diff
@@ -13,42 +13,117 @@
...
+void func_808A7954(ObjWturn* this) {
+    this->actionFunc = func_808A7968;
+}
+
+void func_808A7968(ObjWturn* this, GlobalContext* globalCtx) {
+    if (globalCtx->msgCtx.unk1202A >= 28 && globalCtx->msgCtx.unk1202A < 39) {
+        Actor_UnsetSwitchFlag(globalCtx, this->actor.params);
+        Actor_MarkForDeath(&this->actor);
+    } else if ((Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == 88)) ||
+               (!Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == 89))) {
```
Comment (LtPeriwinkle) [inline src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c:48]: `sceneNum` has an enum, these two are `SCENE_F40` and `SCENE_F41`, and the one later on is also `SCENE_F40`
```suggestion
    } else if ((Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == SCENE_F40)) ||
               (!Flags_GetSwitch(globalCtx, this->actor.params) && (globalCtx->sceneNum == SCENE_F41))) {
```

File: src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c [hunk 2]
```diff
@@ -13,42 +13,117 @@
...
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7AAC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7BA0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7C04.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/func_808A7C78.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Wturn/ObjWturn_Update.s")
+static Vec3f D_808A7DC0 = { 0.0f, -1.0f, 0.0f };
```
Comment (LtPeriwinkle) [inline src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c:31]: This could probably be moved down to near (or inside) the function where it's used.

**Review by LtPeriwinkle — COMMENTED** [review]
Nice job, just two small things I see.

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c [hunk 3]
```diff
@@ -13,42 +13,116 @@
...
+    Player* player = GET_PLAYER(globalCtx);
+
+    this->unk_14A++;
+    player->actor.world.pos.y = this->actor.world.pos.y + this->unk_14A * 4.0f;
+    func_8016981C(globalCtx, this->camId, &player->actor.focus.pos, &camera->eye, &D_808A7DC0);
+    if (this->unk_14A == 1) {
+        globalCtx->unk_1887F = 0x40;
+        gSaveContext.nextTransition = 3;
+        gSaveContext.nextCutsceneIndex = 0;
+        if (globalCtx->sceneNum == 0x58) {
```
Comment (LtPeriwinkle) [inline src/overlays/actors/ovl_Obj_Wturn/z_obj_wturn.c:115]: Sorry missed this one last time, this is also `SCENE_F40` i think

**Review by isghj5 — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]
Jenkins is mad because there is a build error. You can see the output here:
https://jenkins.deco.mp/job/MM/job/PR-564/24/

I believe the recent actor documentation renamed a function, and you missed it in your PR.

**Pull request merged**